### ISSUE TITLE
Addition of gtksourceview-4 breaks builds in multiple platforms

### ISSUE THREAD

**Issue by derbyw** [GtkSharp/GtkSharp#196]
The recent change to add the GtkSourceView 4.0 widget causes the Sample app to fail when run on systems with older/non-existent versions of gtksourceview - i.e. Ubuntu LTS 18.04 has a gtksourceview-3.x version but no apt-get-able version 4 version - the Windows MSYS2 system by default also does not have this installed so as a consequence anyone building in windows or an older Linux system - will have their build fail and will be forced to either build the library locally just to run the sample app.  On my embedded system this will also require me build to the widget and provide/update it's lib requirements, etc. just to run the sample app so I can debug some memory leak issues I am seeing on Eto (See Eto Issue #1796)

It might be nice to clearly indicate this new dependency in the GtkSharp "build instructions" or better, provide some sort of build flag to not require the widget (I don't need control this in my embedded Eto.Forms based app)

**Comment by hol430** [comment]
Installing libgtksourceview4 on windows via msys should be pretty straightforward, but if I recall correctly it shouldn't be necessary - I thought the gtk deps were automatically downloaded as part of the windows build. On Linux you could always build the widget from source...if the build tools are all available. If others agree, we could always just change the sample to use gtktextview instead of gtksourceview. I don't mind too much either way but I do recall seeing a similar issue recently so it sounds like a few people are running into this problem.

**Comment by harry-cpp** [comment]
Nah, sample will use gtksourceview. If you are developing an app, I would assume that you have a proper and up to date development environment.

**Comment by derbyw** [comment]
RE windows build: I thought the gtk deps were automatically downloaded as part of the windows build..

Maybe on some systems - but when I run the app from VS 2019 I get the following:
         DllNotFoundException: GtkSource: libgtksourceview-4-0.dll, libgtksourceview-4.so.0, libgtksourceview-4.0.dylib, gtksourceview-4.dll

> If you are developing an app, I would assume that you have a proper and up to date development environment.
Wow... that's quite an assumption - like there could be no possible valid reason(s) that you would not - and what exactly is proper and up to date?  

This is a recent change and I assumed you would care that in 2 different "normal" environments including Windows 10, the sample app now fails without undocumented mucking about.

All I'm saying is please have the courtesy to future builders of the library by documenting that gtksourceview 4 is a now a build requirement and may require manual installation or what you mean by "proper and up to date development environment".

**Comment by harry-cpp** [comment]
> Wow... that's quite an assumption - like there could be no possible valid reason(s) that you would not - and what exactly is proper and up to date?
> 
> This is a recent change and I assumed you would care that in 2 different "normal" environments including Windows 10, the sample app now fails without undocumented mucking about.
> 
> All I'm saying is please have the courtesy to future builders of the library by documenting that gtksourceview 4 is a now a build requirement and may require manual installation or what you mean by "proper and up to date development environment".

I was only answering the question about not including gtksourceview in the samples, not anything about documentation / updating windows native deps.

Also, please use less aggressive tone.

> Anyways, Im just answering to take the opportunity to ask @harry-cpp the best approach to redistribute a windows application.

Lazy way is grab a zip from https://github.com/GtkSharp/Dependencies and extract it next to your published .exe :)

**Comment by derbyw** [comment]
Apologies for the tone - the second part of your response came across as flippant to me - sorry for overreacting.

**Comment by jwosty** [comment]
Ran into this but was able to fix it by installing the gtksourceview library.
